jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -212,18 +217,24 @@ private void pullBaseImage() {
             new PullBaseImageStep(buildConfiguration, childProgressDispatcherFactory));
   }
 
-  private void pullAndCacheBaseImageLayers() {
+  private void pullAndCacheBaseImageLayers(boolean forcePull) {
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
 
     results.baseImageLayers =
         executorService.submit(
             () ->
                 scheduleCallables(
-                    PullAndCacheBaseImageLayerStep.makeList(
-                        buildConfiguration,
-                        childProgressDispatcherFactory,
-                        results.baseImageAndAuth.get())));
+                    forcePull
+                        ? ObtainBaseImageLayerStep.makeListForForcedDownload(
+                            buildConfiguration,
+                            childProgressDispatcherFactory,
+                            results.baseImageAndAuth.get())
+                        : ObtainBaseImageLayerStep.makeListForSelectiveDownload(
```

Comment:
I don't much like these method names: there's a lot of implicit knowledge required to connect `forcePull` to _selective downloading_.  Can we rename `forcePull` to `layersRequiredLocally`?